File: src/main/java/org/is/bandmanager/service/LocationService.java
package org.is.bandmanager.service;

import jakarta.validation.Valid;
import org.is.bandmanager.dto.LocationDto;
import org.is.bandmanager.dto.request.LocationRequest;
import org.is.bandmanager.model.Location;

import java.util.List;

public interface LocationService {

	LocationDto create(@Valid LocationRequest request);

	List<LocationDto> getAll();

	LocationDto get(Long id);

	Location getEntity(Long id);

	LocationDto update(Long id, @Valid LocationRequest request);

	LocationDto delete(Long id);

}

File: src/app/services/location.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {delay, Observable} from 'rxjs';
import {Location} from '../models/location.model';
import {LocationRequest} from '../models/requests/location-request.model';
import {environment} from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class LocationService {
  private baseUrl = `${environment.apiUrl}/locations`;

  constructor(private http: HttpClient) {
  }

  getAll(): Observable<Location[]> {
    return this.http.get<Location[]>(this.baseUrl).pipe(delay(0));
  }

  get(id: number): Observable<Location> {
    return this.http.get<Location>(`${this.baseUrl}/${id}`);
  }

  create(request: LocationRequest): Observable<Location> {
    return this.http.post<Location>(this.baseUrl, request);
  }

  update(id: number, request: LocationRequest): Observable<Location> {
    return this.http.put<Location>(`${this.baseUrl}/${id}`, request);
  }

  delete(id: number): Observable<Location> {
    return this.http.delete<Location>(`${this.baseUrl}/${id}`);
  }

}
